### PR TITLE
Adding support for gosec scan on CDI repo

### PR DESCRIPTION
**Pull request by visheshtanksale**
**What this PR does / why we need it**:
Adding [gosec](https://github.com/securego/gosec) to the repo.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kubevirt/containerized-data-importer/issues/1615

**Special notes for your reviewer**:

**Release note**:

```release-note
Support to run gosec security scanner
```

### CODE REVIEW

**Comment by brybacki** [comment]
all the lanes failed on: `docker: Error response from daemon: manifest for kubevirt/kubevirt-cdi-bazel-builder:0.0.11 not found: manifest unknown: manifest unknown.`

**Comment by visheshtanksale** [comment]
> all the lanes failed on: `docker: Error response from daemon: manifest for kubevirt/kubevirt-cdi-bazel-builder:0.0.11 not found: manifest unknown: manifest unknown.`
> what might be wrong here?

@brybacki I am updating the builder image to include gosec in it. I have made an update in the dockerfile. How can I get the docker builder image build and pushed with the proper tag to make this work?

**Comment by awels** [comment]
So the order is to create a PR that creates the new builder image with the new tag. Merge that, that will create the new image, then in the follow up PR that uses it, grab the sha256 from the image and use that.

**Comment by visheshtanksale** [comment]
> So the order is to create a PR that creates the new builder image with the new tag. Merge that, that will create the new image, then in the follow up PR that uses it, grab the sha256 from the image and use that.

Thanks @awels

**Comment by visheshtanksale** [comment]
@awels created this PR to generate new builder image https://github.com/kubevirt/containerized-data-importer/pull/1679. Does it look ok?

**Comment by awels** [comment]
Yes, that PR looks okay, I will give it the required acks to merge.

**Comment by visheshtanksale** [comment]
> Yes, that PR looks okay, I will give it the required acks to merge.

Hi @awels, the builder image with new tag is not yet pushed to docker hub. How to initiate pushing of the new builder image?

**Comment by maya-r** [comment]
It looks like quay.io/kubevirt/kubevirt-cdi-bazel-builder:0.0.11 exists.
Perhaps that's the one we should use -- we recently switched to Quay to avoid DockerHub's rate-limiting

File: hack/build/config.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ FUNC_TEST_BAD_WEBSERVER="cdi-func-test-bad-webserver"
 FUNC_TEST_PROXY="cdi-func-test-proxy"
 # update this whenever builder Dockerfile is updated
 BUILDER_TAG=${BUILDER_TAG:-0.0.11}
-BUILDER_IMAGE=${BUILDER_IMAGE:-kubevirt/kubevirt-cdi-bazel-builder@sha256:13de53f8b31aed4b1b66ac9533ab27d29e0dde8d70eba28908673085d9b0fe39
+BUILDER_IMAGE=${BUILDER_IMAGE:-kubevirt/kubevirt-cdi-bazel-builder@sha256:145765dd23572a0bc085f351848f937e2df630f6570794b10041400b47bc0552
```
Comment (awels) [inline hack/build/config.sh:31]: I think it should be:
```sh
BUILDER_IMAGE=${BUILDER_IMAGE:-quay.io/kubevirt/kubevirt-cdi-bazel-builder@sha256:145765dd23572a0bc085f351848f937e2df630f6570794b10041400b47bc0552
```

**Comment by visheshtanksale** [comment]
@awels Can you help with this PR?

**Pull request merged**